### PR TITLE
[SYCLomatic][CMake] Support migration of configure_file command and emit warning msg to users

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/test/dpct/cmake_migration/case_053/expected.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# DPCT3001:1: "configure_file" is used to copy an input file to an output file while performing transformations of the input file content. You may need to copy the input file from CUDA code project to the migrated code project to make sure the command configure_file run successfully or consider removing this command.
```
Comment (zhimingwang36) [inline clang/test/dpct/cmake_migration/case_053/expected.txt]: DPCT3001:1: "configure_file" is used to copy an input file to an output file while performing transformations of the input file content. You may need to update the name of the input and output file if the input file contains CUDA syntax, and make sure the input file exists by copying or migrating the original input file from in-root folder.

Comment (tomflinda) [reply clang/test/dpct/cmake_migration/case_053/expected.txt]: Updated.

**Review by zhimingwang36 — APPROVED** [review]

File: clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml [hunk 2]
```diff
@@ -171,6 +171,8 @@
       In: ${arg}.cpp
       Out: ${arg}.${rewrite_extention_name}
 
+# Please do not modify "CmakeSyntax" field for this rule, as this rule has been hard-coded
```
Comment (zhimingwang36) [inline clang/tools/dpct/DpctOptRules/cmake_script_migration_rule.yaml]: Please fix the hard-code which is not expected.

**Pull request merged**